Extension/src/LanguageServer/extension.ts

```diff
@@ -959,17 +960,64 @@ function reportMacCrashes(): void {
     }
 }
 
-let previousMacCrashData: string;
-let previousMacCrashCount: number = 0;
+export function watchForCrashes(crashDirectory: string): void {
+    if (process.platform !== "win32") {
+        prevCrashFile = "";
+        fs.stat(crashDirectory, (err) => {
+            const crashObject: Record<string, string> = {};
+            if (err?.code) {
+                // If the directory isn't there, we have a problem...
+                crashObject["fs.stat: err.code"] = err.code;
+                telemetry.logLanguageServerEvent("MacCrash", crashObject, undefined);
+                return;
+            }
 
-function logMacCrashTelemetry(data: string): void {
+            // vscode.workspace.createFileSystemWatcher only works in workspace folders.
+            try {
+                fs.watch(crashDirectory, (event, filename) => {
+                    if (event !== "rename") {
+                        return;
+                    }
+                    if (!filename || filename === prevCrashFile) {
+                        return;
+                    }
+                    prevCrashFile = filename;
+                    if (!filename.startsWith("cpptools")) {
+                        return;
+                    }
+                    // Wait 5 seconds to allow time for the crash log to finish being written.
+                    setTimeout(() => {
+                        fs.readFile(path.resolve(crashDirectory, filename), 'utf8', (err, data) => {
+                            void handleCrashFileRead(crashDirectory, filename, err, data);
+                        });
+                    }, 5000);
+                });
+            } catch (e) {
+                // The file watcher limit is hit (may not be possible on Mac, but just in case).
```

Comment:
Do we need to watch for crashes, or can we just check for the existence of a crash dump file when initializing a new language server?  I thought that was how the old mac crashes worked, but it looks like I was wrong.  However, since we control/own this file and are actively deleting it after it's processed, maybe we don't need the watcher?

Reply:
The plan is to add support for cpptools-srv crashes so I think the file watcher allows us to detect/send the crashes sooner and it ensures we know where to look in the temp location returned from initialize.

Reply:
Ok, yeah, we'll want the file watcher to handle cpptools-srv crashes.